webapp/src/utils/jira_issue_metadata.tsx

```diff
@@ -248,3 +248,84 @@ export function isEpicIssueType(issueType: IssueType): boolean {
 export function isMultiSelectField(field: FilterField): boolean {
     return field.schema.type === 'array';
 }
+
+// Some Jira fields have special names for JQL
+function getFieldNameForJQL(field: FilterField) {
+    switch (field.key) {
+    case 'fixVersions':
+        return 'fixVersion';
+    case 'versions':
+        return 'affectedVersion';
+    default:
+        return field.name;
+    }
+}
+
+function quoteGuard(s: string) {
+    if (s && s.includes(' ')) {
+        return `"${s}"`;
+    }
+
+    return s;
+}
+
+export function generateJQLStringFromSubscriptionFilters(issueMetadata: IssueMetadata, fields: FilterField[], filters: ChannelSubscriptionFilters) {
```

Comment:
One more step towards reimplementing Jira, right? :)

Reply:
I can't wait to have to re-implement this server side for some reason 😆